marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/impl/process.xqy

```diff
@@ -335,9 +339,12 @@ declare function proc-impl:process-match-and-merge-with-options(
           => map:get("context")
         let $current-collections := $write-context
           => map:get("collections")
-        let $new-collections := coll-impl:on-no-match(
-          map:entry($uri, $current-collections),
-          $on-no-match
+        let $new-collections := (
```

Comment:
And then "build-collections-for-merged-uri"? Could put this entire block into that function:

```
let $write-context := $write-object
          => map:get("context")
        let $current-collections := $write-context
          => map:get("collections")
        let $new-collections := coll-impl:on-no-match(
          map:entry($uri, $current-collections),
          $on-no-match
        )
        let $_update := $write-context
          => map:put("collections", $new-collections)
```

Reply:
Let's make a new task to refactor this and unit test this.